internal/proxy/add.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+# add.sh adds new dependencies to the Go Cloud module proxy. This script
+# should be run from the root of the repository.
+
+set -o pipefail
+
+function log() {
+	echo "add.sh: $@" 1>&2
+}
+
+function die() {
+	log "$@"
+	exit 1
+}
+
+if ! [[ -d .git ]]; then die 'this script must be run from the root of the repository'; fi
+
+# This approach was suggested by:
+# https://github.com/go-modules-by-example/index/tree/master/012_modvendor
+
+log "creating temp dir where we'll create a module download cache"
+tgp="$(mktemp -d)"
+if [[ $? != 0 ]]; then die 'failed'; fi
+
+temp=$(mktemp)
+if [[ $? != 0 ]]; then die 'failed to create temp file for verbose command output'; fi
+
+# Copy current module cache into temporary directory as basis.
+# Periodically, someone on the project should go through this process without
+# running this step to prune unused dependencies.
+log 'making temp download dir'
+if ! mkdir -p "$tgp/pkg/mod/cache/download"; then die 'failed'; fi
+log "downloading current module cache, logging to $temp"
+if ! gsutil -m rsync -r gs://go-cloud-modules "$tgp/pkg/mod/cache/download" > $temp 2>&1; then die 'failed'; fi
+
+log "filling cache with all module dependencies from current branch, logging to $temp"
+if ! ./internal/proxy/makeproxy.sh "$tgp" > $temp 2>&1; then die 'failed while running makeproxy.sh'; fi
+
+log 'moving the temporary cache to modvendor'
+if ! rm -rf modvendor; then die 'failed to remove modvendor dir'; fi
```

Comment:
Copying the cache to a local directory is helpful for human introspection, but not necessary AFAIK for scripted operation. I believe you can do the `rsync` without ill effects.

Reply:
Done.

Reply:
You may be right about this, but I'm not familiar enough with gsutil rsync and the proxy to be sure it will work with this change. Adding a TODO.

Reply:
Please open a "code health" issue to track it.

Reply:
Done. https://github.com/google/go-cloud/issues/1043